packages/react/src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,93 @@
+import { Slot } from '@radix-ui/react-slot';
+import cl from 'clsx/lite';
+import {
+  Fragment,
+  type HTMLAttributes,
+  type ReactNode,
+  forwardRef,
+  useMemo,
+} from 'react';
+
+export type AvatarProps = {
+  /**
+   * The name of the person the avatar represents.
+   * Will be used to generate initials if no image is provided.
+   */
+  name?: string;
+  /**
+   * The color of the avatar.
+   *
+   * @default 'accent'
+   */
+  color?: 'accent' | 'neutral' | 'brand1' | 'brand2' | 'brand3';
+  /**
+   * The size of the avatar.
+   *
+   * @default 'md'
+   */
+  size?: 'xs' | 'sm' | 'md' | 'lg';
+  /**
+   * The shape of the avatar.
+   *
+   * @default 'circle'
+   */
+  variant?: 'circle' | 'square';
+  /**
+   * Image or icon to display inside the avatar.
+   *
+   * Gets `aria-hidden="true"`
+   */
+  children?: ReactNode;
+} & HTMLAttributes<HTMLSpanElement>;
+
+const fontSizeMap = {
+  xs: 'ds-paragraph--xs',
+  sm: 'ds-heading--2xs',
+  md: 'ds-heading--sm',
+  lg: 'ds-heading--md',
+};
+
+export const Avatar = forwardRef<HTMLSpanElement, AvatarProps>(function Avatar(
+  {
+    name,
+    color = 'accent',
+    size = 'md',
+    variant = 'circle',
+    className,
+    children,
+    ...rest
+  },
+  ref,
+) {
+  const Component = children ? Slot : Fragment;
+
+  const initials = useMemo(() => {
+    return getInitials(name);
+  }, [name]);
+
+  return (
+    <span
+      ref={ref}
+      className={cl('ds-avatar', fontSizeMap[size], className)}
```

Comment:
Should we wait with the font-size, and rather fix this in CSS with composes instead of adding as separate class now? ☺️

Reply:
I wrote this in the description: "Font class will be removed later in favor of composes in modules".
We need this to get all properties from the font style, and not just the size.